### DISCUSSION TITLE
How can i start a new blockchain? (was: Blockscout has corrupted my hardhat localhost node)

### DISCUSSION THREAD

**Discussion by oehm-smith** [NomicFoundation/hardhat#4796] [Troubleshooting]
Hi all,

I raised an issue about Blockscout at https://ethereum.stackexchange.com/questions/159305/blockscout-hardhat-unable-to-find-a-block-containing-transaction.  I've returned to my smart contracts project after some days away and a restart of my Mac and even without Blockscout running I am seeing the same continuous stream of debug messages after running `npx hardhat node`:

```
...
debug_traceTransaction

  Unable to find a block containing transaction 0x06a63101cdb516b4fab23acbbb9f6585fc22e6e51cde42c388417c27cd188878

debug_traceTransaction

  Unable to find a block containing transaction 0x4ebd04d62bdb384f61a6a2d08fdbdceb441e6b976bae01e845987ad34b774a6b

debug_traceTransaction

  Unable to find a block containing transaction 0xa4176d2900a4004b0c472a3f50a3fb94a7b660b32834ede0f3b632d8d87b984c

debug_traceTransaction

  Unable to find a block containing transaction 0x4111eef31bc23dfe6eb2d553eee4aba8194b6300231ef11a6fc5dc278b262bc4

debug_traceTransaction

  Unable to find a block containing transaction 0x42dc578ad6261a5927401225f01b64f331c7f6cafbbdddedeef16b0138b6f0de

debug_traceTransaction

  Unable to find a block containing transaction 0x06a63101cdb516b4fab23acbbb9f6585fc22e6e51cde42c388417c27cd188878

...
```

So it seems the blockchain somehow became corrupt.

I tried a `npx hardhat clean` and it only cleans up the artifacts it seems and not the local blockchain.

How can I restart a fresh blockchain or otherwise clean this up.

Any opinions on the Blockscout problem welcome also.

**Comment by oehm-smith** [comment]
I'm guessing from the lack of answers that this is impossible or not easy to do.

**Reply by fvictorio** [reply]
I'm not sure what is the question tbh. If you start a fresh Hardhat node and a fresh blockscout instance, then the behavior shouldn't fundamentally change. If you restart the Hardhat node without restarting the blockscout instance, I would expect errors like this to pop up.

My guess is that you have some sort of process still running and trying to reconcile data from a previous node run with a new one, or something like that.

**Reply by oehm-smith** [reply]
I thought I was also seeing this when Blockscout wasn't running.  I'm mistaken and only see this when BS IS running.  Still, I'd like to run it so would like to work out this problem.

I looked through the code and there is no obvious way to switch off this debug messsge.  The root of this is in geth and there is https://geth.ethereum.org/docs/interacting-with-geth/rpc/ns-debug#debugtracetransaction but that doesn't help me get rid of it.

I had to grep it out so I could see the log output:

```
npx hardhat --tsconfig ./tsconfig.hardhat.json node | grep -v -e "debug_traceTransaction" -e "Unable to find a block containing transaction" | sed '/^$/d'
```

I would like to try and confirm blockscout (BS) as the cause or something happened as a coincidence when i started using BS.  So I would like to start with a clean blockchain and accounts.

Can anyone tell me how to do that?

----

After running it this way (grepping out the junk) and not having all those debug trace messages, I'm now seeing the following.  Quite a few of these also (but nowhere nearrrrr as many as the debug trace I have been discussing).

Also only occurring when BlockScout is running:

```
Received invalid block tag 4. Latest block number is 0
```

I'm seeing tag 3,2,1,4,13 in that order over an over. 

```
eth_getBlockByNumber
eth_getBlockByNumber
eth_getBlockByNumber
eth_getBlockByNumber (2)
eth_getBlockByNumber
eth_getCode
  Received invalid block tag 3. Latest block number is 0
eth_getCode
  Received invalid block tag 2. Latest block number is 0
eth_getCode
  Received invalid block tag 1. Latest block number is 0
eth_getCode
  Received invalid block tag 4. Latest block number is 0
eth_getCode
  Received invalid block tag 13. Latest block number is 0
eth_getBalance
  Received invalid block tag 3. Latest block number is 0
eth_getBalance
  Received invalid block tag 2. Latest block number is 0
eth_getBalance
  Received invalid block tag 1. Latest block number is 0
eth_getBalance
  Received invalid block tag 4. Latest block number is 0
eth_getBalance
  Received invalid block tag 13. Latest block number is 0
eth_getBlockByNumber
```

The zero (0) would imply it is a clean blockchain, right?

And if so then what state exists such that it is receiving invalid block tags?  Where does that come from?  Is this message related?  Is it an error or expected for some reason?

I'm going to grep these out also as they are taking up too much vertical space.

I am also piping through `uniq` to remove the repeated adjacent lines.  I don't think I'll try and uniq adjacent blocks :)

```
eth_getCode
eth_getBalance
eth_getBlockByNumber
eth_getBlockByNumber (2)
eth_getBlockByNumber
eth_getCode
eth_getBalance
eth_getBlockByNumber
eth_getBlockByNumber (2)
eth_getBlockByNumber
eth_getCode
eth_getBalance
eth_getBlockByNumber
eth_getBlockByNumber (2)
eth_getBlockByNumber
````

I'm not cranky, just trying to understand and learn :)

**Reply by oehm-smith** [reply]
Mmmm, maybe I should try a new fresh clean BlockScout DB.  It must have held on to transactions and is querying those ...

**Reply by oehm-smith** [reply]
Nope makes no difference:

```
cd blockscout
docker-compose rm -f
docker-compose -f hardhat-network.yml up -d
```

And still seeing lots of these (many different txs)

```
debug_traceTransaction

  Unable to find a block containing transaction 0x7422fee7a6af5a6a6cf7a31f3f5ad9c6d42e03d43563b62bed9fe170e2a29bb4
```

But that transaction exists and a block number is displayed.  And the TX was created 2 weeks ago so the blockchain isn't starting clean with a new node.

And if i drill down in to the block there is one tx - 0x7422fee7a6af5a6a6cf7a31f3f5ad9c6d42e03d43563b62bed9fe170e2a29bb4

![2024-02-06 Hardat debug_traceTransaction](https://github.com/NomicFoundation/hardhat/assets/2556595/b60db5ca-4470-4ed9-9e0d-55600c713b55)

So I'm leaning back towards this is a Hardhat issue.

Any help on how to debug this problem?  Or my original request for details no how to start a clean blockchain?

**Reply by fvictorio** [reply]
Make sure that you are actually removing the blockscout volume. `docker-compose` is confusing in that respect. I don't remember exactly what command you have to use; I think it was something like `docker-compose down -v`. Obviously, be careful when you do this, since it can remove important data.
